src/java/htsjdk/samtools/cram/build/CramContainerIterator.java

```diff
@@ -51,6 +52,7 @@ public boolean hasNext() {
 
     @Override
     public Container next() {
+        if(!hasNext()) { throw new NoSuchElementException(); }
```

Comment:
Here's another example of how iterators were broken, before this change, calling next() twice in a row would return null in the second time...